server/routes/activity.js

```diff
@@ -1,15 +1,17 @@
 const Router = require('koa-router');
 const Activity = require('../models/activity');
 const validateActivity = require('../middleware/validation/validateActivity');
+const queryStringSearch = require('../middleware/queryStringSearch');
+
 
 const router = new Router({
   prefix: '/activity'
 });
 
 
 
-router.get('/:id', async ctx => {
-  const activity = await Activity.query().where('id', ctx.params.id);
+router.get('/:id', queryStringSearch, async ctx => {
+  const activity = await Activity.query().where(ctx.query);
```

Comment:
The queries need to be on the naked route, don't they? `/:id` will only return a single result, can't narrow that down with a query...

Reply:
I realised that and I'm currently working on putting them on the naked route.